lib/ansible/utils/vars.py

```diff
@@ -79,12 +79,12 @@ def _validate_mutable_mappings(a, b):
         )
 
 
-def combine_vars(a, b):
+def combine_vars(a, b, merge=False):
     """
     Return a copy of dictionaries of variables based on configured hash behavior
     """
 
-    if C.DEFAULT_HASH_BEHAVIOUR == "merge":
+    if merge or C.DEFAULT_HASH_BEHAVIOUR == "merge":
```

Comment:
i would make distinction here, make it `merge=None` above and allow user to be explicit and when `None` fallback to the existing config

`if merge or merge is None and C.DEFAULT_HASH_BEHAVIOUR == "merge":`

This allows for explicit overrides both ways, at least while the configuration exists.

Reply:
Yeah, that's better. Done.